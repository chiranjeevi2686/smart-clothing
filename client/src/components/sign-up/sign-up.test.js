import React from "react";
import { render, fireEvent } from '@testing-library/react';
import SignUp from './sign-up.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const defaultStore =
{
    displayName: '',
    name: '',
    email: '',
    password: '',
    confirmPassword: ''
};
const mockStore = configureStore();
describe('SignUp component', () => {
    it('should render SignUp', async () => {
        const { asFragment, getByTestId } = render(<Provider store={mockStore(defaultStore)}>
            <SignUp />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
        const displayName = getByTestId('display-name')
        const signUpEmail = getByTestId('sign-up-email')
        const signUpPwd = getByTestId('sign-up-pwd')
        const signUpCnfPwd = getByTestId('sign-up-cnf-pwd')
        const signUnButton = getByTestId('sign-up-btn')
        fireEvent.change(displayName, { target: { value: 'test' } })
        fireEvent.change(signUpEmail, { target: { value: 'testemail' } })
        fireEvent.change(signUpPwd, { target: { value: 'testpwd' } })
        fireEvent.change(signUpCnfPwd, { target: { value: 'testpwd' } })
        fireEvent.click(signUnButton)

    })
})
