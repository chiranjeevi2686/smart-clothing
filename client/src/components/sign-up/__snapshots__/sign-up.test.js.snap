// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component should render SignUp 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ eNxuqQ"
  >
    <h2
      class="sc-gsTCUz gnBPzW"
    >
      I do not have an account
    </h2>
    <span>
      Sign up with email and password
    </span>
    <form>
      <div
        class="sc-dlfnbm tXewj"
      >
        <input
          class="sc-hKgILt puxKR"
          data-test-id="display-name"
          name="displayName"
          required=""
          type="text"
          value=""
        />
        <label
          class="sc-eCssSg iCNWHH"
        >
          Display Name
        </label>
      </div>
      <div
        class="sc-dlfnbm tXewj"
      >
        <input
          class="sc-hKgILt puxKR"
          data-test-id="sign-up-email"
          name="email"
          required=""
          type="email"
          value=""
        />
        <label
          class="sc-eCssSg iCNWHH"
        >
          Email
        </label>
      </div>
      <div
        class="sc-dlfnbm tXewj"
      >
        <input
          class="sc-hKgILt puxKR"
          data-test-id="sign-up-pwd"
          name="password"
          required=""
          type="password"
          value=""
        />
        <label
          class="sc-eCssSg iCNWHH"
        >
          Password
        </label>
      </div>
      <div
        class="sc-dlfnbm tXewj"
      >
        <input
          class="sc-hKgILt puxKR"
          data-test-id="sign-up-cnf-pwd"
          name="confirmPassword"
          required=""
          type="password"
          value=""
        />
        <label
          class="sc-eCssSg iCNWHH"
        >
          Confirm Password
        </label>
      </div>
      <button
        class="sc-jSgupP wXXsH"
        data-test-id="sign-up-btn"
        type="submit"
      >
        SIGN UP
      </button>
    </form>
  </div>
</DocumentFragment>
`;
